toolchain/check/import_cpp.cpp

```diff
@@ -190,34 +206,45 @@ static auto GenerateAst(Context& context, llvm::StringRef importing_file_path,
                         llvm::ArrayRef<Parse::Tree::PackagingNames> imports,
                         llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs,
                         llvm::StringRef target)
-    -> std::pair<std::unique_ptr<clang::ASTUnit>, bool> {
-  CarbonClangDiagnosticConsumer diagnostics_consumer(&context);
+    -> std::pair<std::unique_ptr<InFlightClang>, bool> {
+  auto generated_imports_h =
+      (importing_file_path + ".generated.cpp_imports.h").str();
+  fs = OverlayFile(fs, generated_imports_h,
+                   GenerateCppIncludesHeaderCode(context, imports));
 
   // TODO: Share compilation flags with ClangRunner.
-  auto ast = clang::tooling::buildASTFromCodeWithArgs(
-      GenerateCppIncludesHeaderCode(context, imports),
+  std::string target_storage = target.str();
+  const char* args[]{
       // Parse C++ (and not C).
-      {
-          "-x",
-          "c++",
-          // Propagate the target to Clang.
-          "-target",
-          target.str(),
-          // Require PIE. Note its default is configurable in Clang.
-          "-fPIE",
-      },
-      (importing_file_path + ".generated.cpp_imports.h").str(), "clang-tool",
-      std::make_shared<clang::PCHContainerOperations>(),
-      clang::tooling::getClangStripDependencyFileAdjuster(),
-      clang::tooling::FileContentMappings(), &diagnostics_consumer, fs);
-  // Remove link to the diagnostics consumer before its deletion.
-  ast->getDiagnostics().setClient(nullptr);
+      "clang-tool", "-x", "c++",
+      // Do not emit compiler version metadata.
+      "-Qn",
+      // Flags to match LLVM IR output of default code generation options, which
+      // was used in an earlier version.
+      // TODO: remove them when sharing flags with ClangRunner.
+      "-fomit-frame-pointer", "-fno-unwind-tables",
```

Comment:
I am **really** unsure if I should be adding this and it should definitely go away after the TODO about sharing with ClangRunner is finished.

Happy to take either path here and not sure which defaults Carbon uses now.

Reply:
Regarding your question about keeping output the same: if you want to decouple this PR from changes to the output, one way to do that would be to overwrite the `CodeGenOptions` in the `CompilerInvocation` with a default-constructed `CodeGenOptions` for now (and just set the `EmitVersionIdentMetadata` flag to false). For things that we want full control over, changing them via `CompilerInvocation` seems better than trying to control them with driver arguments.